lib/tictactoe.rb

```diff
@@ -26,63 +19,72 @@ def print_board()
 
   def play_game
     @move_count = 0
-    @current_player = nil
-    while @move_count < 10 && !won?
-      @current_player = @move_count.even? ? player1 : player2
-      print "#{@current_player.name} Enter a row: "
-      row = gets.chomp.to_i
+    current_player = nil
+    has_won = false
+
+    while !draw? && !has_won
+      current_player = @move_count.even? ? player1 : player2
+
+      row = get_input(current_player, 'row').to_i
       next unless valid_input?(row)
 
-      print 'Enter a column: '
-      column = gets.chomp.to_i
+      column = get_input(current_player, 'column').to_i
       next unless valid_input?(column)
 
       if board.move_valid?(row - 1, column - 1)
-        board.apply_coords(row - 1, column - 1, @current_player.sign)
+        board.apply_coords(row - 1, column - 1, current_player.sign)
         @move_count += 1
+        puts @move_count
       end
+
       print_board
+      has_won = diagonal?(current_player.sign) || axial?(current_player.sign)
     end
-    game_result
+    puts draw? ? "It's a draw" : "#{current_player.name} has won"
   end
 
-  def game_result
-    puts 'Result:'
-    if won?
-      puts "#{@current_player.name} has won"
-    else
-      puts 'Its a draw'
+  def get_input(player, coordinate)
+    print "#{player.name} enter a #{coordinate}: "
+    begin
+      Integer(gets)
+    rescue StandardError
+      nil
     end
   end
 
   def valid_input?(input)
-    valid = input >= 1 && input <= 3
+    valid = input >= 1 && input <= 3 # checks whether input is between 1 and 3 inclusively
     puts 'invalid input' unless valid
```

Comment:
- [ ] This should be in the main. You can just return instead of printing and print in the main file